java/runtime/src/main/java/org/ray/runtime/raylet/MockRayletClient.java

```diff
@@ -82,14 +89,53 @@ public UniqueId generateTaskId(UniqueId driverId, UniqueId parentTaskId, int tas
 
   @Override
   public <T> WaitResult<T> wait(List<RayObject<T>> waitFor, int numReturns, int timeoutMs) {
-    return new WaitResult<T>(
-        waitFor,
-        ImmutableList.of()
-    );
+
+      List<UniqueId> ids = new ArrayList<>();
+      for (RayObject<T> element : waitFor) {
+          ids.add(element.getId());
+      }
+      boolean[] ready = new boolean[ids.size()];
+      try{
+          Future future = exec.submit(() -> {
+              int trueCount = 0;
+              while (true){
+                  for (int i = 0; i < ids.size(); i++) {
+                      if ( !ready[i] && store.isObjectReady(ids.get(i))) {
+                          ready[i] = true;
+                          ++trueCount;
+                      }
+                      if(trueCount == numReturns){
+                          return;
+                      }
+                  }
+              }
+          });
+          future.get(timeoutMs,TimeUnit.MILLISECONDS);
```

Comment:
`future.get` is blocking. so, this is equivalent to sleep timeoutMs and check. right?

Reply:
future.get will wait for at most the given time for the computation to complete, and then retrieves its result, if available.  the result can return before timeout if it's ready